modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/SupportedLanguageManager.java

```diff
@@ -0,0 +1,138 @@
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.translation.exception.TranslatorException;
+import com.liferay.translation.translator.deepl.internal.configuration.DeepLTranslatorConfiguration;
+import com.liferay.translation.translator.deepl.internal.model.SupportedLanguage;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Yasuyuki Takeo
+ */
+@Component(immediate = true, service = SupportedLanguageManager.class)
+public class SupportedLanguageManager {
+
+	public List<String> getSupportedLanguages(
+			DeepLTranslatorConfiguration deepLTranslatorConfiguration)
+		throws PortalException {
+
+		try {
+			List<String> languages = new ArrayList<>();
+
+			List<SupportedLanguage> supportedLanguages = toSupportedLanguages(
+				deepLTranslatorConfiguration.authKey(), "target",
+				deepLTranslatorConfiguration.validateLanguageUrl());
+
+			supportedLanguages.forEach(
+				supportedLanguage -> languages.add(
+					supportedLanguage.getLanguage()));
+
+			return languages;
+		}
+		catch (IOException ioException) {
+			_log.error(
+				"Failed to call supported language list." +
+					System.lineSeparator() + ioException.getLocalizedMessage());
+
+			return Collections.emptyList();
+		}
+	}
+
+	protected String getSupportedLanguage(
+			String authKey, String target, String url)
+		throws IOException, PortalException {
+
+		Http.Options options = new Http.Options();
+
+		options.setLocation(
+			URLBuilder.create(
+				url
+			).addParameter(
+				"auth_key", authKey
+			).build());
+
+		options.addHeader(
+			HttpHeaders.CONTENT_TYPE,
+			ContentTypes.APPLICATION_X_WWW_FORM_URLENCODED);
+		options.addPart("auth_key", authKey);
+		options.addPart("target", target);
+		options.setMethod(Http.Method.POST);
+
+		String supportedLanguage = _http.URLtoString(options);
+
+		Http.Response response = options.getResponse();
+
+		Response.Status status = Response.Status.fromStatusCode(
+			response.getResponseCode());
+
+		if (status == Response.Status.OK) {
+			return supportedLanguage;
+		}
+
+		throw new TranslatorException(
+			"The status is " + status + ". Please retry after a while.");
+	}
+
+	protected List<SupportedLanguage> toSupportedLanguages(
+			String authKey, String target, String url)
+		throws IOException, PortalException {
+
+		JSONArray jsonArray = _jsonFactory.createJSONArray(
+			getSupportedLanguage(authKey, target, url));
+
+		return JSONUtil.toList(
+			jsonArray,
+			customFieldJSONObject -> new SupportedLanguage() {
+				{
+					language = customFieldJSONObject.getString("language");
+					name = customFieldJSONObject.getString("name");
+					supportsFormality = customFieldJSONObject.getBoolean(
+						"supports_formality");
+				}
+			},
```

Comment:
Don't use a dynamic init. block. Add a constructor to SupportedLanguage that takes the necessary parameters.